src/Microsoft.OData.Core/UriParser/SemanticAst/ReferenceSegment.cs

```diff
@@ -0,0 +1,79 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ReferenceSegment.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.UriParser
+{
+    #region Namespaces
+    using Microsoft.OData.Edm;
+    #endregion Namespaces
+
+    /// <summary>
+    /// A segment representing $ref in a path.
+    /// </summary>
+    public sealed class ReferenceSegment : ODataPathSegment
+    {
+        /// <summary>
+        /// Build a segment representing $ref.
+        /// </summary>
+        /// <param name="navigationSource">The navigation source to which this segment references.</param>
+        /// <exception cref="System.ArgumentNullException">Throws if any input parameter is null.</exception>
+        public ReferenceSegment(IEdmNavigationSource navigationSource)
+        {
+            ExceptionUtils.CheckArgumentNotNull(navigationSource, "navigationSource");
+
+            this.Identifier = UriQueryConstants.RefSegment;
+            this.SingleResult = false;

```

Comment:
>this.SingleResult = false; [](start = 12, length = 26)

what does setting this to false mean?  Can I still do Employees(1)/Manager/$ref?

Reply:
From looking at the ODataPathParser, it looks like we throw if the source is a single result. I don't recall why we do that. If we decide that this is should not be the case, then I can re-write to accept single results.

```
if (this.parsedSegments.Last().TargetKind != RequestTargetKind.Resource || this.parsedSegments.Last().SingleResult)
{
  throw ExceptionUtil.CreateBadRequestError(ODataErrorStrings.PathParser_EntityReferenceNotSupported(this.parsedSegments.Last().Identifier));
}

ReferenceSegment referenceSegment = new ReferenceSegment(lastNavigationSource);
```

Reply:
Resolved offline; will allow single results and base SingleResult off of whether navigationSource is a collection.